### PR TITLE
samples: basic: blink_led: add support PWM for the stm32f4_disco

### LINKED ISSUE #4760: stm32f4_disco and frdm_k64f  samples/basic/blink_led ,Choose supported PWM driver
```
/Volumes/Embedded/Zephyr/zephyr/samples/basic/blink_led/src/main.c:33:2: error: #error "Choose supported PWM driver"
 #error "Choose supported PWM driver"
  ^
/Volumes/Embedded/Zephyr/zephyr/samples/basic/blink_led/src/main.c: In function 'main':
/Volumes/Embedded/Zephyr/zephyr/samples/basic/blink_led/src/main.c:50:31: error: 'PWM_DRIVER' undeclared (first use in this function)
  pwm_dev = device_get_binding(PWM_DRIVER);
                               ^
/Volumes/Embedded/Zephyr/zephyr/samples/basic/blink_led/src/main.c:50:31: note: each undeclared identifier is reported only once for each function it appears in
/Volumes/Embedded/Zephyr/zephyr/samples/basic/blink_led/src/main.c:57:33: error: 'PWM_CHANNEL' undeclared (first use in this function)
   if (pwm_pin_set_usec(pwm_dev, PWM_CHANNEL,
                                 ^
make[3]: *** [src/main.o] Error 1
make[2]: *** [zephyr-app-dir] Error 2
make[1]: *** [sub-make] Error 2
make: *** [all] Error 2

```

### PR DESCRIPTION
**Pull request by zhang-wenchao**
Fix #4760
Signed-off-by: Zhang WenChao <zhangwenchao001@gmail.com>

### CODE REVIEW

**Comment by spring-z** [comment]
@galak @erwango 
Sorry.thank you for your help

**Comment by spring-z** [comment]
[YouTube Video](https://www.youtube.com/watch?v=nCQjuAUB0wU)

**Review by erwango — CHANGES_REQUESTED** [review]
Fix is ok.

Miss some doc:
-Update samples/basic/blink_led/README.rst

Then split the change into 2 commits:
*board: Add PWM2 on stm32f4_disco
*sample: blink_led: Add support for stm32f4_disco

Then ok for merge

**Comment by spring-z** [comment]
successful check @erwango

File: samples/basic/blink_led/README.rst [hunk 1]
```diff
@@ -22,8 +22,8 @@ Arduino 101 and Quark D2000 CRB
 You will need to connect the LED to ground and PWM0 via the shield.
 You may need a current limiting resistor. See your LED datasheet.
 
-Nucleo_F401RE and Nucleo_L476RG
-===============================
+Nucleo_F401RE and Nucleo_L476RG and STM32F4_DISCOVERY
```
Comment (dbkinder) [inline samples/basic/blink_led/README.rst]: change to:

    Nucleo_F401RE, Nucleo_L476RG, and STM32F4_DISCOVERY

**Review by dbkinder — CHANGES_REQUESTED** [review]
A little fix to the heading and otherwise looks good.

**Comment by spring-z** [comment]
Already modified, thanks. @dbkinder

**Review by dbkinder — APPROVED** [review]

**Comment by spring-z** [comment]
@erwango

**Review by galak — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**